src/Message/SmsMessageResult.php

```diff
@@ -48,6 +48,13 @@ class SmsMessageResult implements SmsMessageResultInterface {
   public $reports;
 
   /**
+   * The messages associated with this result.
```

Comment:
> The initial model was for message results to be a property of an SMS message

When was this model in?

> there's a bit of recursivity and data duplication here (especially significant with large number of recipients).

Im not sure entirely what you mean by this. The messages are added to the result after they are `sent()`/`incoming()`. At least they are not recycled again, so PHP engine will not create new objects internally (should be same memory usage as well).

If you have ideas....